server/src/error.rs

```diff
@@ -0,0 +1,26 @@
+use core::{
+    error::Error,
+    fmt::{Debug, Display},
+};
+
+use crate::platform::{crypto::Drbg, TpmContextDeps};
+
+pub enum ServerError<Deps: TpmContextDeps> {
+    Drbg(<Deps::Drbg as Drbg>::Error),
+}
+
+impl<Deps: TpmContextDeps> Debug for ServerError<Deps> {
```

Comment:
Why not just `#[derive(Debug)]` on the error type?

Reply:
As I've stated in other comments, I don't think we should even have a generic over `Deps`. But I'll leave it to you.